api/gpt.py

```diff
@@ -33,11 +38,23 @@ class GPT:
 
     def __init__(self, engine='davinci',
                  temperature=0.5,
-                 max_tokens=100):
+                 max_tokens=100,
+                 input_prefix = "input: ",
+                 input_suffix = "\n",
+                 output_prefix = "output: ",
+                 output_suffix = "\n",
+                 append_input_suffix_and_output_prefix_to_query = False,
+                 stop = "\ninput:"):
         self.examples = []
         self.engine = engine
         self.temperature = temperature
         self.max_tokens = max_tokens
+        self.input_prefix = input_prefix
+        self.input_suffix = input_suffix
+        self.output_prefix = output_prefix
+        self.output_suffix = output_suffix
+        self.append_input_suffix_and_output_prefix_to_query = append_input_suffix_and_output_prefix_to_query
+        self.stop = stop
 
     def add_example(self, ex):
```

Comment:
In this class, you could include a method such as:

```
def format_example(self, ex):
        """Formats the input, output pair."""
        return self.input_prefix + ex.get_input() + self.input_suffix + self.output_prefix + ex.get_output() + self.output_suffix
```

Then the `add_example` method could be:

```
def add_example(self, ex):
        """Adds an example to the object. Example must be an instance
        of the Example class."""
        assert isinstance(ex, Example), "Please create an Example object."
        self.examples.append(self.format_example(ex))
```

Reply:
format_example method added which is now called by add_example as suggested

Reply:
👍